pkg/cfn/builder/vpc.go

```diff
@@ -35,6 +47,26 @@ func (c *ClusterResourceSet) addSubnets(refRT *gfn.Value, topology api.SubnetTop
 			SubnetId:     refSubnet,
 			RouteTableId: refRT,
 		})
+
+		if api.IsEnabled(c.spec.VPC.AutoAllocateIPv6) {
+			// get 8 of /64 subnets from the auto-allocated IPv6 block,
+			// and pick one block based on subnetIndexForIPv6 counter;
+			// NOTE: this is done inside of CloudFormation using Fn::Cidr,
+			// we don't slice it here, just construct the JSON expression
+			// that does slicing at runtime.
+			refAutoAllocateCIDRv6 := gfn.MakeFnSelect(
```

Comment:
Since this is inside a big loop and these are quite low level details, can we encapsulate lines 57-66 into a function like `addCIDR6Selector(subnetIndex)`

Reply:
Well, the puzzling thing is that it's all a bit tied into how the outer function works, don't you think? I'd much rather take all of the conditional IPv6 bits out, but that asks for more refactoring...

Reply:
I've tried, but it turns more complicated for what its worth... I'd rather live it for later.